lib/streamlit/runtime/app_session.py

```diff
@@ -489,14 +486,6 @@ def _on_secrets_file_changed(self, _) -> None:
         # `_on_source_file_changed` just for this purpose sounded finicky.
         self._on_source_file_changed()
 
-    def _on_pages_changed(self, _) -> None:
-        msg = ForwardMsg()
-        self._populate_app_pages(msg.pages_changed, self._pages_manager.get_pages())
```

Comment:
it looks like the signature of `_populate_app_pages` then could also be changed from

```python
def _populate_app_pages(
        self, msg: NewSession | PagesChanged, pages: dict[PageHash, PageInfo]
    ) -> None:
```
to

```python
def _populate_app_pages(
        self, msg: NewSession, pages: dict[PageHash, PageInfo]
    ) -> None:
```